packages/devtools_app/lib/src/flutter/split.dart

```diff
@@ -100,29 +101,70 @@ class _SplitState extends State<Split> {
     final height = constraints.maxHeight;
     final axisSize = isHorizontal ? width : height;
     final crossAxisSize = isHorizontal ? height : width;
-    const halfDivider = Split.dividerMainAxisSize / 2.0;
 
     // Determine what fraction to give each child, including enough space to
     // display the divider.
-    double firstSize = axisSize * firstFraction;
-    double secondSize = axisSize * secondFraction;
-
-    // Clamp the sizes to be sure there is enough space for the dividers.
-    firstSize = firstSize.clamp(halfDivider, axisSize - halfDivider);
-    secondSize = secondSize.clamp(halfDivider, axisSize - halfDivider);
+    final numDividers = widget.children.length - 1;
+    final sizes = List.generate(
+      fractions.length,
+      (i) =>
+          (axisSize - numDividers * Split.dividerMainAxisSize) * fractions[i],
+    );
 
-    // Remove space from each child to place the divider in the middle.
-    firstSize = firstSize - halfDivider;
-    secondSize = secondSize - halfDivider;
+    void updateSpacing(DragUpdateDetails dragDetails, int splitterIndex) {
+      final dragDelta =
+          isHorizontal ? dragDetails.delta.dx : dragDetails.delta.dy;
+      final fractionalDelta = dragDelta / axisSize;
+
+      double updateSpacingBeforeSplitterIndex(double delta) {
+        var index = splitterIndex;
+        while (index >= 0) {
+          fractions[index] += delta;
+          if (fractions[index] >= 0.0) {
+            _clampFraction(index);
+            return 0.0;
+          }
+          delta = fractions[index];
+          _clampFraction(index);
+          index--;
+        }
+        return delta;
+      }
+
+      double updateSpacingAfterSplitterIndex(double delta) {
```

Comment:
nit: the return value from these methods doesn't seem like it is used. Make it void if it isn't used.